modules/library/main/src/main/java/org/geotools/styling/SLDParser.java

```diff
@@ -1713,7 +1713,17 @@ protected Mark parseMark(Node root) {
             } else if (childName.equalsIgnoreCase("WellKnownName")) {
                 if (LOGGER.isLoggable(Level.FINEST))
                     LOGGER.finest("setting mark to " + getFirstChildValue(child));
-                mark.setWellKnownName(parseCssParameter(child));
+                Expression wellKnownName = parseCssParameter(child);
+                if (wellKnownName instanceof Literal) {
+                    String expanded = wellKnownName.evaluate(null, String.class);
+                    if (expanded != null && expanded.startsWith("file://")) {
+                        URL url = onlineResourceLocator.locateResource(expanded);
+                        if (url != null) {
+                            wellKnownName = ff.literal(url.toExternalForm());
+                        }
```

Comment:
If the resource is not found here it will never be found right ? I would a log warning.

Reply:
Yep, doing so